connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1123,7 +1123,7 @@ class SparkConnectPlanner(session: SparkSession) {
         val combinedUnion = CombineUnions(
           Union(
             Seq(transformRelation(u.getLeftInput), transformRelation(u.getRightInput)),
-            byName = u.getByName))
+            byName = u.getByName, allowMissingCol = u.getAllowMissingColumns))
```

Comment:
`assert(!allowMissingCol || byName, "`allowMissingCol` can be true only if `byName` is true.")`

Union itself does the assert on this. Because it is an assert so I think we should check here and throw proper exception back to client (so not depend on the assert).

Reply:
@amaliujia added a `InvalidPlanInput` exception and a negative test case